### PR TITLE
[4.0][workflow] - installation with sample_testing data

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #21564 .

### Summary of Changes
populate the `#__workflow_associations` table with items from `#__content` table

### Testing Instructions
fresh install staging
Customize installation

skip languages  &Install sample data

### Expected result
all contents are visible on Articles Manager

### Actual result
no content

### CODE REVIEW

**Comment by infograf768** [comment]
This will not solve the issue for installing sample data via the sample data module.

**Comment by alikon** [comment]
yes correct this is only about the install with sample data
for the  the sample data module i'm looking at your pr #21553

**Comment by infograf768** [comment]
My POC forces the use of a hardcoded value in __workflows_associations because I want these articles Published. It came from the default stage id 2 for the workflow_id 1 (Joomla! Default).
Therefore it is very complex to solve really as it would depend on the default workflow when using the sample data module.

This PR here is also using a hardcoded value but it may work because we have only ONE default workflow at install time. (Not sure btw what has been decided with installing sample data at Joomla install time).

**Comment by brianteeman** [comment]
Except that currently you can install sample data at anytime

**Comment by alikon** [comment]
At install time there is and must be only 1 workflow the Joomla! Default one,
plus in the update.sql script we are assuming the same 2=published Joomla! Default stage
so hardcoding at install time ie with sample_testing.sql should be safe

one problem at the time please

**Comment by Bakual** [comment]
The sampledata and its SQL files are supposed to be removed from the installation. There should only be the module for installing sampledata and that is done after the installation from the control panel.

So not sure if this PR is really needed.

**Comment by alikon** [comment]
> The sampledata and its SQL files are supposed to be removed from the installation.

good to know 

just want to know where me/we (mere mortals) can read about this Breaking News
And/Or when  this should happen

**Comment by brianteeman** [comment]
In dont think that removing the ability to install sql files from th installer is likely to happen as they are used heavily made template developers in their quick start builds

**Comment by Bakual** [comment]
> just want to know where me/we (mere mortals) can read about this Breaking News
And/Or when this should happen

The one and single goal of the "Sample Data Module" was to be able to drop the sample data SQL files and the sample data step in the installation. Maintaining both the sample data plugin(s) and the SQL files would be quite stupid.
We can't drop it in J3, but we (hopefully) will in J4.

> In dont think that removing the ability to install sql files from th installer is likely to happen as they are used heavily made template developers in their quick start builds

I'm not familiar with how exactly templates add SQLs to their builds. Was that done in the same sample data step? Can't they use a sample data plugin to set up their example settings?

**Comment by brianteeman** [comment]
> The one and single goal of the "Sample Data Module" was to be able to drop the sample data SQL files and the sample data step in the installation. Maintaining both the sample data plugin(s) and the SQL files would be quite stupid.

Adding the plugin enabled users who installed joomla through a script provided by their hosts the opportunity to get the sample data

> I'm not familiar with how exactly templates add SQLs to their builds. Was that done in the same sample data step?

Quite often yes - I know yootheme does.

Basically in J3 if you drop an sql file into the installation/sql folder then the installer would pick it up

>  Can't they use a sample data plugin to set up their example settings?

I guess so - but its quite hard to write isnt it (especially if you have a lot of data)?

**Comment by wojsmol** [comment]
@Bakual Additionally the sample data plugin for testing sample data is now incompatible with Joomla! 4, see https://github.com/joomla-extensions/testing-sample-data/issues/2.

**Comment by Bakual** [comment]
> Basically in J3 if you drop an sql file into the installation/sql folder then the installer would pick it up

Ah, didn't know that was dynamically generated. If we need to still support that, it could be done so it only shows that step if there actually are any files and we remove the sample data ones? Never tried what happens if there are no files at all.

> I guess so - but its quite hard to write isnt it (especially if you have a lot of data)?

Depends what you want to achieve, It can be done very simple: You could make it applying still an SQL file. It gets complex once you want to support different languages and use entered data in a successive step like in the testing sampledata where an article refers to another article.

**Comment by Bakual** [comment]
> Additionally the sample data plugin for testing sample data is now incompatible with Joomla! 4, 

Just needs someone who creates a PR and someone else who tests it and then someone else who merges it 😄

**Comment by wojsmol** [comment]
@Bakual I can test but needed changes in code are to hard for me to make :smile:

**Comment by Bakual** [comment]
It's hard to fix anyway as J4 is still a moving target.

**Comment by wojsmol** [comment]
I know that but for me the namespacing part is hard.

File: installation/src/Model/DatabaseModel.php [hunk 1]
```diff
@@ -1212,4 +1216,37 @@ protected function splitQueries($query)
...
+		$workflow = new Workflow(['extension' => 'com_content']);
+		$query = $db->getQuery(true);
+		$query->select($db->quoteName('id'))
+			->from($db->quoteName('#__content'));
+
+		$db->setQuery($query);
+
+		try
+		{
+			$contents = $db->loadObjectList();
```
Comment (bembelimen) [inline installation/src/Model/DatabaseModel.php:1240]: Perhaps ```->loadColumn()``` and act with the array alone?

**Comment by bembelimen** [comment]
I have tested this item :white_check_mark: successfully on 95c4c9e3f8435c9bd3f873dd2e43d599ca8331b3<br /><br />Looks good<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21681">issues.joomla.org/tracker/joomla-cms/21681</a>.</sub>

**Comment by alikon** [comment]
too much mess

**Pull request closed without merging**